### ISSUE TITLE
[Bug] Total number of rule should be known ahead of time in the logs, and should be static

### ISSUE THREAD

**Issue by hhpatel14** [konveyor/analyzer-lsp#434]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Konveyor version

N/A

### Priority

Critical

### Current Behavior

The analysis log displays the total number of rules used during the analysis. The progress of the analysis can be monitored by tracking additional variables, such as the number of rules that have failed, matched, and remained unmatched, against the total number of rules. The total number of rules should be known in advance and remain constant throughout the analysis process.

### Expected Behavior

Total number of rules should be static and displayed before analysis. 

### How Reproducible

Always (Default)

### Environment

```markdown

```

**Comment by jmle** [comment]
@hhpatel14 this sounds more like a feature request than a bug. What do you think?

**Comment by jwmatthews** [comment]
This issue is related to work to support the IDE Plugins and matching behavior with WindUp.

WindUp had functionality to print out the current rule / total rule as @hhpatel14 mentioned and the IDE plugins use that to build functionality to inform the user of progress.  As @hhpatel14 is updating the IDE Plugins to work with Kantra she ran into not having this functionality to rely on.

For immediate needs, I assume (needs to be verified) we can skip monitoring fine-grained progress of Kantra from the IDE Plugins, so for that reason I would assume we can lower the priority of this request so it doesn't need to be 'critical', yet I think it's a valid issue to address when we have the bandwidth.
